src/components/cc-popover/cc-popover.js

```diff
@@ -0,0 +1,264 @@
+import '../cc-button/cc-button.js';
+import { css, html, LitElement } from 'lit';
+import { createRef, ref } from 'lit/directives/ref.js';
+import { dispatchCustomEvent, EventHandler } from '../../lib/events.js';
+
+/**
+ * @typedef {import('../common.types.js').IconModel} IconModel
+ * @typedef {import('./cc-popover.types.js').PopoverPosition} PopoverPosition
+ */
+
+/**
+ * A component displaying a floating content next to a button element.
+ *
+ * ## Details
+ *
+ * ### Button element
+ *
+ * The `button` is the element that will trigger the display of the floating content.
+ * This element is a `cc-button`. The `cc-button:click` event will trigger the popover toggle.
+ *
+ * ### Popover content
+ *
+ * The `content` is the element that will be displayed next to the `button` element.
+ * This element must be placed in the default slot.
+ * It will be placed according to the `position` property.
+ *
+ * ## Accessibility
+ *
+ * The component places necessary aria attribute on the `button` element:
+ *
+ * * `aria-expanded`: set to `true` when popover is open, `false` when it is closed.
+ *
+ * When the popover is open, pressing `esc` will close the popover and focus the `button` element.
+ *
+ * ## Usage
+ *
+ * ```html
+ * <cc-popover>
+ *   <span slot="button-text">Click me</span>
+ *   <div>This is a content to be displayed when button is clicked.</div>
+  * </cc-popover>
+ * ```
+ *
+ * @cssdisplay block
+ *
+ * @event {CustomEvent} cc-popover:open - Fires whenever the popover is opened.
+ * @event {CustomEvent} cc-popover:close - Fires whenever the popover is closed.
+ *
+ * @slot - The area containing the content of the popover.
+ * @slot button - The area containing the button content.
+ *
+ * @cssprop {Size} --cc-popover-gap - Sets the gap between the button and the floating area (default 0.4em).
+ * @cssprop {Number} --cc-popover-z-index - Sets the z-index of the floating content (defaults: `999`).
+ */
+export class CcPopover extends LitElement {
+  static get properties () {
+    return {
+      accessibleName: { type: String, attribute: 'accessible-name' },
+      hideText: { type: Boolean, attribute: 'hide-text' },
+      icon: { type: Object },
+      isOpen: { type: Boolean, attribute: 'is-open', reflect: true },
+      position: { type: String },
+    };
+  }
+
+  constructor () {
+    super();
+
+    /** @type {string|null} Sets the accessibleName property of the underlying `cc-button` element. CAUTION: The accessible name should always start with the visible text if there is one. For instance "add to estimation - NodeJS XS" */
```

Comment:
question: `For instance "add to estimation - NodeJS XS".` Is this not relevant here?